### PR TITLE
fix download manager

### PR DESCRIPTION
**Pull request by anhpham2403**
fix download manager and musicservice

### CODE REVIEW

File: app/src/main/java/com/framgia/soundcloud/data/model/User.java [hunk 1]
```diff
@@ -8,7 +8,18 @@
  * Created by anh on 23/10/2017.
  */
 
-public class User implements Parcelable{
+public class User implements Parcelable {
```
Comment (DoanVanToan) [inline app/src/main/java/com/framgia/soundcloud/data/model/User.java:11]: WARNING: [ShortClassName] Avoid short class names like User
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/naming.html#ShortClassName

File: app/src/main/res/values/dimens.xml [hunk 2]
```diff
@@ -7,4 +7,5 @@
     <dimen name="sp_15">15sp</dimen>
     <dimen name="sp_13">13sp</dimen>
     <dimen name="sp_18">18sp</dimen>
+    <dimen name="dp_100">100dp</dimen>
```
Comment (DoanVanToan) [inline app/src/main/res/values/dimens.xml:10]: WARNING: [UnusedResources] The resource `R.dimen.dp_100` appears to be unused
 Unused resources make applications larger and slow down builds.

Comment (anhpham2403) [reply app/src/main/res/values/dimens.xml:10]: Em dùng cho pull sau ạ.

File: app/src/main/java/com/framgia/soundcloud/service/MusicService.java [hunk 3]
```diff
@@ -225,15 +229,21 @@ public void setLoop(@LoopMode int loop) {
...
     /**
      * loop mode
      */
     @Retention(SOURCE)
     @IntDef({ LoopMode.NONE_LOOP, LoopMode.LOOP_LIST_TRACKS, LoopMode.LOOP_TRACK })
     public @interface LoopMode {
-    int NONE_LOOP = 0;
-    int LOOP_LIST_TRACKS = 1;
-    int LOOP_TRACK = 2;
+        int NONE_LOOP = 0;
```
Comment (DoanVanToan) [inline app/src/main/java/com/framgia/soundcloud/service/MusicService.java]: ERROR: 'member def type' have incorrect indentation level 8, expected level should be 4.

File: app/src/main/java/com/framgia/soundcloud/service/MusicService.java [hunk 4]
```diff
@@ -225,15 +229,21 @@ public void setLoop(@LoopMode int loop) {
...
+        int LOOP_LIST_TRACKS = 1;
```
Comment (DoanVanToan) [inline app/src/main/java/com/framgia/soundcloud/service/MusicService.java]: ERROR: 'member def type' have incorrect indentation level 8, expected level should be 4.

File: app/src/main/java/com/framgia/soundcloud/service/MusicService.java [hunk 5]
```diff
@@ -225,15 +229,21 @@ public void setLoop(@LoopMode int loop) {
...
+        int LOOP_TRACK = 2;
```
Comment (DoanVanToan) [inline app/src/main/java/com/framgia/soundcloud/service/MusicService.java]: ERROR: 'member def type' have incorrect indentation level 8, expected level should be 4.

File: app/src/main/java/com/framgia/soundcloud/service/MusicService.java [hunk 6]
```diff
@@ -245,10 +255,10 @@ public void setLoop(@LoopMode int loop) {
             StateMode.STATE_STOP
     })
     public @interface StateMode {
-    int STATE_IDLE = 0;
-    int STATE_PLAYING = 1;
-    int STATE_PAUSE = 2;
-    int STATE_STOP = 3;
+        int STATE_IDLE = 0;
```
Comment (DoanVanToan) [inline app/src/main/java/com/framgia/soundcloud/service/MusicService.java]: ERROR: 'member def type' have incorrect indentation level 8, expected level should be 4.

File: app/src/main/java/com/framgia/soundcloud/service/MusicService.java [hunk 7]
```diff
@@ -245,10 +255,10 @@ public void setLoop(@LoopMode int loop) {
...
+        int STATE_PLAYING = 1;
```
Comment (DoanVanToan) [inline app/src/main/java/com/framgia/soundcloud/service/MusicService.java]: ERROR: 'member def type' have incorrect indentation level 8, expected level should be 4.

File: app/src/main/java/com/framgia/soundcloud/service/MusicService.java [hunk 8]
```diff
@@ -245,10 +255,10 @@ public void setLoop(@LoopMode int loop) {
...
+        int STATE_PAUSE = 2;
```
Comment (DoanVanToan) [inline app/src/main/java/com/framgia/soundcloud/service/MusicService.java]: ERROR: 'member def type' have incorrect indentation level 8, expected level should be 4.

File: app/src/main/java/com/framgia/soundcloud/service/MusicService.java [hunk 9]
```diff
@@ -245,10 +255,10 @@ public void setLoop(@LoopMode int loop) {
...
+        int STATE_STOP = 3;
```
Comment (DoanVanToan) [inline app/src/main/java/com/framgia/soundcloud/service/MusicService.java]: ERROR: 'member def type' have incorrect indentation level 8, expected level should be 4.

File: app/src/main/AndroidManifest.xml [hunk 10]
```diff
@@ -5,6 +5,8 @@
     <uses-permission android:name="android.permission.INTERNET"/>
     <uses-permission android:name="android.permission.WAKE_LOCK"/>
     <uses-permission android:name="android.permission.WRITE_EXTERNAL_STORAGE"/>
+    <uses-permission android:name="android.permission.READ_EXTERNAL_STORAGE"/>
```
Comment (DoanVanToan) [inline app/src/main/AndroidManifest.xml]: Khi yeeu cau cap quyen WRITE_EXTERNAL_STORAGE da co quyen READ_EXTERNAL_STORAGE 
https://developer.android.com/reference/android/Manifest.permission.html

File: app/src/main/java/com/framgia/soundcloud/service/MusicService.java [hunk 11]
```diff
@@ -225,6 +229,12 @@ public void setLoop(@LoopMode int loop) {
         }
     }
 
+    @Override
+    public void onBufferingUpdate(MediaPlayer mp, int percent) {
+        double ratio = percent / 100.0;
```
Comment (DoanVanToan) [inline app/src/main/java/com/framgia/soundcloud/service/MusicService.java]: harrdcode

**Review by DoanVanToan — APPROVED** [review]

**Pull request merged**